---
title: "Itâ€™s all about the zeros: estimating stochastic zeros in shallow scRNA-seq experiments"
output: html_document
author: Timothy Daley
---

In shallow single cell RNA-seq (scRNA-seq), the problem of zeros is pervasive.  There are two main sources of zeros: genes that are not expressed and genes that are expressed but not sequenced.  These are called systematic and stochastic zeros, respectively ([Lun et al 2016](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0947-7)).  Stochastic zeros can arise from two sources: genes that were unsequenced due to the shallow sequencing depth and genes that dropped out due to technical effects.  Distinguishing between the two is difficult.

Consider the following model proposed by [Hicks et al 2015](http://www.biorxiv.org/content/biorxiv/early/2015/12/27/025528.full.pdf).  Let $\lambda_{ij}$ be the true unobserved expression level for gene $j$ in cell (or sample) $i$.   Let $Z_{ij}$ be an indicator variable denoting whether or not gene $j$ was amplified in cell $i$.  The observed expression $X_{ij}$ then has expectation equal to
$$\mathrm{E} (X_{ij}) = Z_{ij} \lambda_{ij}.$$
I will assume that conditional up $Z_{ij} = 1$ then $X_{ij}$ is a Poisson random variable with expectation $\lambda_{ij}$ and that the $\lambda$s are iid distributed from some underlying compounding distribution $\mu$. This model includes other models such as the Negative Binomial model described by [Lun et al 2016](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0947-7). Additionally I will assume (following Hicks et al) that $Z_{ij}$ is a Bernoulli random variable that depends only on the cell, i.e. $Pr(Z_{ij} = 1) = p_{i}$.

<!-- 
Suppose we observe expression $x_{ij} > 0$.  What is the expectation of $\lambda_{ij}$ conditional upon the non-zero observed expression level?  Note that if $x_{ij} > 0$ then necessarily $Z_{ij} = 1$.
$$\mathrm{E} (\lambda_{ij} | x_{ij} = x, Z_{ij = 1}) = \frac{\mathrm{E} (\lambda \cdot 1(X_{ij} = x_{ij}) \cdot 1(Z_{ij} = 1))}{\Pr(X_{ij} = x_{ij}, Z_{ij} = 1)} = \frac{\int_{0}^{\infty} \lambda \frac{\lambda^{x} e^{-\lambda}}{x!} p_{i} d \mu (\lambda)}{\int_{0}^{\infty} \frac{\lambda^{x} e^{- \lambda}}{x!} p_{i} d \mu (\lambda)} = p_{i} \frac{(x + 1) \Pr(X = x + 1)}{\Pr(X = x)}.$$

Now suppose we observed expression $x_{ij} = 0$.  What is the expectation of $\lambda_{ij}$ conditional upon the zero observed expression level?  In this case we do not know if the expression level is zero because of insufficient sequencing depth (or Poisson random noise) or if $Z_{ij} = 0$.  I will assume that if $Z_{ij} = 0$ then $\lambda_{ij} = 0$.
$$\mathrm{E} (\lambda_{ij} | x_{ij} = 0) = \mathrm{E}(\lambda_{ij} | x_{ij} = 0) \Pr(Z_{ij} = 1) + \mathrm{E}(\lambda_{ij} | x_{ij}) \Pr(Z_{ij} = 0) = p_{i} \frac{\int_{0}^{\infty} \lambda e^{-\lambda} d \mu (\lambda)}{\int_{0}^{\infty} e^{- \lambda} d \mu (\lambda)} = p_{i} \frac{Pr(X = 1)}{\Pr(X = 0)}.$$ 
-->

# Estimating $p_{i}$

Estimating $p_{i}$ is difficult because one can not seperate the stochastic zeros from the true zeros.  The estimate of $p_{i}$ will inherently be biased by sequencing depth, as increased sequencing depth will mean a decrease in stochastic zeros.  This is a problem in the current state of scRNA-seq as the field moves more towards low coverage sequencing of more cells ([Macosko et al 2015](http://www.sciencedirect.com/science/article/pii/S0092867415005498), [Shekhar et al 2016](http://www.sciencedirect.com/science/article/pii/S0092867416310078), [Heimberg et al 2016](http://www.sciencedirect.com/science/article/pii/S2405471216301090)).  

The main problem is to estimate the number of genes with positive expression but were missed by random chance.  This is a familiar problem in the capture-recapture literature, as we can see the immediate connection to such problems as estimating the number of words Shakespeare knew but are not in his writings [Efron and Thisted 1976](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.460.3131&rep=rep1&type=pdf), estimating the unobserved T-Cell $\beta$ repertoire [Qi et al 2014](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.460.3131&rep=rep1&type=pdf), or estimating the number of unobserved genomic variants in a population [Gravel 2014](http://www.genetics.org/content/197/2/601.short) and [Zou et al 2016](http://www.nature.com/articles/ncomms13293).  The main takeaway from these problems is that when sampling (sequencing in our case) is insufficiently deep then the observed number of genes is not a very good estimate of the number of expressed genes.  (Example here).
